rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala

```diff
@@ -231,54 +233,58 @@ object Runtime {
   private val MATCH_UNLIMITED_PHLOS = matchListPar(Cost(Integer.MAX_VALUE))
 
   private def introduceSystemProcesses(
-      space: RhoISpace,
-      replaySpace: RhoISpace,
-      processes: immutable.Seq[(Name, Arity, Remainder, BodyRef)]
-  ): Seq[Option[(TaggedContinuation, Seq[ListParWithRandomAndPhlos])]] =
-    processes.flatMap {
-      case (name, arity, remainder, ref) =>
-        val channels = List(name)
-        val patterns = List(
-          BindPattern(
-            (0 until arity).map[Par, Seq[Par]](i => EVar(FreeVar(i))),
-            remainder,
-            freeCount = arity
-          )
-        )
-        val continuation = TaggedContinuation(ScalaBodyRef(ref))
-        Seq(
-          space.install(channels, patterns, continuation)(MATCH_UNLIMITED_PHLOS),
-          replaySpace.install(channels, patterns, continuation)(MATCH_UNLIMITED_PHLOS)
-        )
-    }
+      space: RhoISpace[Task],
+      replaySpace: RhoISpace[Task],
+      processes: Seq[(Name, Arity, Remainder, BodyRef)]
+  ): Task[Seq[Option[(TaggedContinuation, Seq[ListParWithRandomAndPhlos])]]] =
+    Task.sequence(
+      processes
+        .flatMap {
+          case (name, arity, remainder, ref) =>
+            val channels = List(name)
+            val patterns = List(
+              BindPattern(
+                (0 until arity).map[Par, Seq[Par]](i => EVar(FreeVar(i))),
+                remainder,
+                freeCount = arity
+              )
+            )
+            val continuation = TaggedContinuation(ScalaBodyRef(ref))
+            Seq(
+              space.install(channels, patterns, continuation)(MATCH_UNLIMITED_PHLOS),
+              replaySpace.install(channels, patterns, continuation)(MATCH_UNLIMITED_PHLOS)
+            )
+        }
+    )
```

Comment:
don't you think that it would be beneficial to keep these methods abstract (`F`)? Task is not enforced here.